source/util/FileUtils.cpp

```diff
@@ -66,7 +66,13 @@ string FileUtils::ExtractExpandedPath(const string &filePath)
         return "";
     }
     wordexp_t word;
-    wordexp(filePath.c_str(), &word, 0); // TODO: How to handle error?
+    int status = wordexp(filePath.c_str(), &word, 0); // TODO: How to handle error?
+    if (status)
+    {
+        throw Aws::Iot::DeviceClient::Util::wordexp_fail_error(
+            "Received status from wordexp: " + std::to_string(status) +
+            "\nPertaining to following filepath: " + filePath);
```

Comment:
No embedded newlines in the message since this will mess up the logging which prepends a severity (`DEBUG|INFO|WARN|ERROR`) to every line in the log file.